### PR TITLE
Remove openfast_cpp target if BUILD_OPENFAST_CPP_API not selected

### PR DESCRIPTION
**Pull request by andrew-platt**
Ready to merge

**Feature or improvement description**
If `BUILD_OPENFAST_CPP_API` is false and `make` or `make all` is used to build, the `openfast_cpp` target is built.  This can pose issues on some systems if the user is not expecting it and doesn't have locations to all necessary libraries specified.

Also noticed that the `NodeClusterType` input for OpenFOAM was inconsistently handled as integers or logicals.  Changed it all to integer to maintain consistency with checks used in the code.

**Impacted areas of the software**
Interfacing to SOWFA or other CFD.
CMake default builds

**Additional supporting information**
@mchurchf ran into compile issue on Eagle.

**Test results, if applicable**
None.

### CODE REVIEW

File: glue-codes/openfast/CMakeLists.txt [hunk 1]
```diff
@@ -33,5 +30,15 @@ if (${_compiler_id} STREQUAL "GNU" AND NOT ${VARIABLE_TRACKING})
   set_source_files_properties(src/FAST_Prog.f90 PROPERTIES COMPILE_FLAGS "-fno-var-tracking -fno-var-tracking-assignments")
 endif()
 
-install(TARGETS openfast openfast_cpp
+install(TARGETS openfast
   RUNTIME DESTINATION bin)
+
+if(BUILD_OPENFAST_CPP_API AND BUILD_OPENFAST_CPP_DRIVER)
```
Comment (deslaughter) [inline glue-codes/openfast/CMakeLists.txt:36]: `openfast_cpp_driver`doesn't use the `openfastcpplib`, does it need to check for `BUILD_OPENFAST_CPP_API`?

**Review by deslaughter — COMMENTED** [review]
Were you also going to wrap `openfastcpp` in `BUILD_OPENFAST_CPP_DRIVER`? If so, the conditional for regression test `of_cpp_interface_regression` will also need to be updated.

File: glue-codes/openfast/CMakeLists.txt [hunk 2]
```diff
@@ -33,5 +30,15 @@ if (${_compiler_id} STREQUAL "GNU" AND NOT ${VARIABLE_TRACKING})
...
+  add_executable(openfast_cpp_driver src/FAST_Prog.cpp src/FastLibAPI.cpp)
+  target_link_libraries(openfast_cpp_driver openfastlib)
+  
+  install(TARGETS openfast_cpp_driver
+    RUNTIME DESTINATION bin)
+endif(BUILD_OPENFAST_CPP_API AND BUILD_OPENFAST_CPP_DRIVER)
```
[the remark this answers is not in the document]
Comment (andrew-platt) [reply glue-codes/openfast/CMakeLists.txt:42]: Sorry, old habits ;)

**Review by deslaughter — APPROVED** [review]

**Comment by deslaughter** [comment]
Looking back at this, I find it a little strange that `FastLibAPI.h` and `FastLibAPI.cpp` are part of the glue code. It seems like they should be in `openfast-library` and create a C++ library like `openfastlib_cpp` to provide access to OpenFAST from C++. This doesn't need to be changed, just something to consider in the future.

**Pull request merged**